great_tables/_formats.py

```diff
@@ -2215,26 +2222,36 @@ def fmt_datetime_context(
     if is_na(data._tbl_data, x):
         return x
 
-    # From the date and time format strings, create a datetime format string
-    datetime_format_str = f"{date_format_str}'{sep}'{time_format_str}"
+    if format_str is not None:
+        # If `x` is a string, assume it is an ISO datetime string and convert
+        # it to a datetime object
+        if isinstance(x, str):
+            # Convert the ISO datetime string to a datetime object
+            x = _iso_str_to_datetime(x)
 
-    # If `x` is a string, assume it is an ISO datetime string and convert it to a datetime object
-    if isinstance(x, str):
-        # Convert the ISO datetime string to a datetime object
-        x = _iso_str_to_datetime(x)
+        x_formatted = x.strftime(format_str)
 
     else:
-        # Stop if `x` is not a valid datetime object
-        _validate_datetime_obj(x=x)
```

Comment:
AFAICT this if / else should just stay the same, and the extra formatting logic should go below it